hudi-common/src/main/java/org/apache/hudi/common/model/HoodieColumnRangeMetadata.java

```diff
@@ -30,16 +28,21 @@
   private final String columnName;
   private final T minValue;
   private final T maxValue;
-  private final long numNulls;
-  private final PrimitiveStringifier stringifier;
+  private long numNulls;
+  // For Decimal Type/Date Type, minValue/maxValue cannot represent it's original value.
+  // eg: when parquet collects column information, the decimal type is collected as int/binary type.
+  // so we cannot use minValue and maxValue directly, use minValueAsString/maxValueAsString instead.
+  private final String minValueAsString;
+  private final String maxValueAsString;
 
-  public HoodieColumnRangeMetadata(final String filePath, final String columnName, final T minValue, final T maxValue, final long numNulls, final PrimitiveStringifier stringifier) {
+  public HoodieColumnRangeMetadata(final String filePath, final String columnName, final T minValue, final T maxValue, long numNulls, final String minValueAsString, final String maxValueAsString) {
     this.filePath = filePath;
     this.columnName = columnName;
     this.minValue = minValue;
     this.maxValue = maxValue;
-    this.numNulls = numNulls;
-    this.stringifier = stringifier;
+    this.numNulls = numNulls == -1 ? 0 : numNulls;
```

Comment:
here in which case would `numNulls == -1` ?

Reply:
When parquet cannot collect statistics for a certain field. For example, parquet does not collect statistics information for timestamp type.   how about throw exception directly？

Reply:
so in this case, the numNulls will be -1 ? and what's the impact if we set numNulls to 0, will the query performance be affected?

Reply:
If this value is - 1, it must be set to 0, This will result in incorrect query results. In fact, when a timestamp type is encountered in subsequent logic, an exception will be thrown directly to tell the user that indexing for timestamp is not supported.

Reply:
If we can not support index timestamp yet, we would throw an exception explicitly

Reply:
yes, we have already throw exceptions in  L264 in ZCurveOptimizeHelper.getMinMaxValue